tests/integration/_experimental/search/test_embeddings.py

```diff
@@ -100,17 +100,18 @@ def dataset_name() -> str:
     ],
 )
 def test__upload_dataset_embeddings(embedding: np.ndarray, dataset_name: str) -> None:
-    upload_dataset_embeddings(
+    _upload_dataset_embeddings(
```

Comment:
Are we testing this internal method because we do not want to surface the `run_embedding_reduction` arg publicly yet?

Reply:
this is causing issues due to different embedding length was uploaded for the same dataset. Also we don't want to trigger an aws job unnecessarily every time an integration test runs.

Reply:
Yup, I'm just wondering why we are testing this internal function instead of the public one, and it seems like they are identical except that the internal one can now take this extra arg that we do not yet want to surface publicly?